pkg/app/client/retry/retry_test.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package retry
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/cloudwego/hertz/pkg/common/test/assert"
+)
+
+func TestApply(t *testing.T) {
+	options := []Option{}
+	option := WithMaxAttemptTimes(100)
+	options = append(options, option)
+	option = WithInitDelay(time.Second)
+	options = append(options, option)
+	option = WithMaxDelay(time.Second)
+	options = append(options, option)
+
+	delayPolicyFunc := func(attempts uint, err error, retryConfig *Config) time.Duration {
+		return time.Second
+	}
+	option = WithDelayPolicy(delayPolicyFunc)
+	options = append(options, option)
+	option = WithMaxJitter(time.Second)
+	options = append(options, option)
+
+	config := Config{}
+	config.Apply(options)
+
+	assert.DeepEqual(t, uint(100), config.MaxAttemptTimes)
```

Comment:
Why only check the MaxAttemptTimes here? Since you've put multiple options before.